integration_test/tests/mongosqltranslate_tests.rs

```diff
@@ -0,0 +1,157 @@
+        let (env_handle, dbc, stmt) = default_setup_connect_and_alloc_stmt(
+            AttrOdbcVersion::SQL_OV_ODBC3,
+            Some(crate::common::generate_srv_style_connection_string(Some(
+                "sample_airbnb".to_string(),
+            ))),
+        );
+
+        unsafe {
+            let mut query: Vec<WideChar> = cstr::to_widechar_vec("SELECT property_type, room_type, bed_type, minimum_nights, maximum_nights FROM listingsAndReviews LIMIT 3");
+            query.push(0);
+
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLExecDirectW(stmt as HStmt, query.as_ptr(), SQL_NTS),
+                "{}",
+                get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, stmt as Handle)
+            );
+
+            get_column_attributes(stmt as Handle, 5);
+
+            fetch_and_get_data(
+                stmt as Handle,
+                Some(3),
+                vec![SqlReturn::SUCCESS; 5],
+                vec![CDataType::SQL_C_WCHAR; 5],
+            );
+
+            disconnect_and_close_handles(dbc, stmt);
+        }
+        let _ = unsafe { Box::from_raw(env_handle) };
+    }
+
+    #[test]
+    fn test_enterprise_mode_with_library_loaded_and_invalid_query_and_valid_schemas_created() {
+        let (env_handle, dbc, stmt) = default_setup_connect_and_alloc_stmt(
+            AttrOdbcVersion::SQL_OV_ODBC3,
+            Some(crate::common::generate_srv_style_connection_string(Some(
+                "sample_airbnb".to_string(),
+            ))),
+        );
+
+        unsafe {
+            let mut query: Vec<WideChar> =
+                cstr::to_widechar_vec("select * from non_existent_collection");
+            query.push(0);
+
+            assert_eq!(
+                SqlReturn::ERROR,
+                SQLPrepareW(stmt as HStmt, query.as_ptr(), SQL_NTS),
+                "{}",
+                get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, stmt as Handle)
+            );
+
+            disconnect_and_close_handles(dbc, stmt);
+        }
+        let _ = unsafe { Box::from_raw(env_handle) };
+    }
+
+    #[test]
+    fn test_enterprise_mode_with_library_loaded_and_valid_query_and_no_sql_schemas_collection() {
+        let (env_handle, dbc, stmt) = default_setup_connect_and_alloc_stmt(
+            AttrOdbcVersion::SQL_OV_ODBC3,
+            Some(crate::common::generate_srv_style_connection_string(Some(
+                "test".to_string(),
+            ))),
+        );
+
+        unsafe {
+            let mut query: Vec<WideChar> = cstr::to_widechar_vec("select * from foo");
+            query.push(0);
+
+            assert_eq!(
+                SqlReturn::ERROR,
+                SQLPrepareW(stmt as HStmt, query.as_ptr(), SQL_NTS),
+                "{}",
+                get_sql_diagnostics(HandleType::SQL_HANDLE_STMT, stmt as Handle)
+            );
+
+            disconnect_and_close_handles(dbc, stmt);
+        }
+        let _ = unsafe { Box::from_raw(env_handle) };
+    }
+}
```

Comment:
One test I see in the design document that I don't see implemented yet is `Invalid Command`.

Reply:
The `Invalid Command` tests ended up being unit tests in `mongosqltranslate`. Since we control the API and the driver's calls to the API, I don't think it's possible to have an integration test that has an invalid command.